src/AWS.Deploy.ServerMode.Client/Utilities/CappedStringBuilder.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace AWS.Deploy.ServerMode.Client.Utilities
+{
+    public class CappedStringBuilder
+    {
+        public int LineLimit { get; set; }
+        public int LineLength { get; private set; } = 0;
+
+        private readonly StringBuilder _stringBuilder;
```

Comment:
I would get rid of the string builder and have a queue of lines of strings that you concat together whenever the ToString method is called.

Reply:
Done